### PR TITLE
Simulation enhancement

### LINKED ISSUE #21: [FEATURE] Update Simulation
### Is your feature request related to a problem? Please describe

The simulation is slow. Files are large. The robot sometimes has seizures and translational motion causes some rotation.

### Describe the solution you'd like

Update the simulation to use smaller, lower-poly files. Update drag/COM to have better movement. Prioritize performance and high quality motion over detailed robot and props.

### PR DESCRIPTION
**Pull request by NoahMollerstuen**
# Pull Request

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Update simulation to GZ Sim Harmonic and improve simulation model and environment

## Related Tickets & Documents

- Related Issue #
- Closes #21

### CODE REVIEW

File: src/surface/rov_gazebo/launch/sim_launch.py [hunk 1]
```diff
@@ -20,9 +20,7 @@ def generate_launch_description() -> LaunchDescription:
     params_path = str(Path(rov_gazebo_path) / 'config' / params_file)
 
     # TODO: gz_sim launch might be nice
-    start_gazebo = ExecuteProcess(
-        cmd=['ign', 'gazebo', '-v', '3', '-r', world_path], output='screen'
-    )
+    start_gazebo = ExecuteProcess(cmd=['gz', 'sim', '-v', '3', '-r', world_path], output='screen')
```
Comment (InvincibleRMC) [inline src/surface/rov_gazebo/launch/sim_launch.py:23]: Did you check out gz_sim launch?

**Review by InvincibleRMC — CHANGES_REQUESTED** [review]

**Comment by InvincibleRMC** [comment]
also remove xacro dep

File: src/surface/rov_gazebo/models/props/coral_restoration_area/model.sdf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+              </geometry>
+            </visual>
+
+            <!-- <collision name="base_link_collision">
+              <geometry>
+                <mesh>
+                  <uri>model://coral_restoration_area/meshes/coral_restoration_area.dae</uri>
+                </mesh>
+              </geometry>
+          </collision> -->
```
Comment (benjaminwp18) [inline src/surface/rov_gazebo/models/props/coral_restoration_area/model.sdf:23]: Did you mean to disable collision here (and on other props)?

File: src/surface/rov_gazebo/models/rov24/model.sdf [hunk 3]
```diff
@@ -5,37 +5,36 @@
...
         </inertia>
       </inertial>
 
       <visual name="base_link_visual">
+        <pose>0.0 0.0 0.0 0.0 0.0 -1.571</pose>
         <geometry>
           <mesh>
             <uri>model://rov24/meshes/rov24.dae</uri>
-            <scale>0.025 0.025 0.025</scale>
+            <!-- <scale>0.0254 0.0254 0.0254</scale> -->
```
Comment (benjaminwp18) [inline src/surface/rov_gazebo/models/rov24/model.sdf:27]: Commented out?

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/surface/rov_gazebo/launch/sim_launch.py (hunk 1 above)
Comment (NoahMollerstuen) [reply src/surface/rov_gazebo/launch/sim_launch.py:23]: Hmm I didn't know this existed but it looks pretty nice
https://index.ros.org/p/ros_gz_sim/

**Review by InvincibleRMC — APPROVED** [review]

File: src/surface/rov_gazebo/scripts/ardupilot_gazebo.sh [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
 sudo apt-get update
-sudo apt-get install rapidjson-dev libignition-gazebo6-dev -y
-# Can't be ssh clone because that breaks docker.
-# Can either make a second file or just hope that our fork does not need lots of dev work
-git clone -b fortress  https://github.com/CWRUbotix/rov-ardupilot-gazebo-fortress.git ~/ardupilot_gazebo
+sudo apt-get install gz-harmonic -y
+sudo apt install libopencv-dev libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev gstreamer1.0-plugins-bad gstreamer1.0-libav gstreamer1.0-gl -y
+wget -q -O - https://raw.githubusercontent.com/mavlink/mavros/master/mavros/scripts/install_geographiclib_datasets.sh | sudo bash
+
+git clone --depth 1 https://github.com/ArduPilot/ardupilot_gazebo  ~/ardupilot_gazebo
```
Comment (benjaminwp18) [inline src/surface/rov_gazebo/scripts/ardupilot_gazebo.sh:12]:
```suggestion
git clone --depth 1 https://github.com/ArduPilot/ardupilot_gazebo ~/ardupilot_gazebo
```

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

**Pull request merged**